backend/tests/integration/pidstore/minters/test_minters_bai.py

```diff
@@ -0,0 +1,56 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2019 CERN.
+#
+# inspirehep is free software; you can redistribute it and/or modify it under
+# the terms of the MIT License; see LICENSE file for more details.
+
+
+import pytest
+from helpers.providers.faker import faker
+from invenio_pidstore.models import PersistentIdentifier, PIDStatus
+
+from inspirehep.pidstore.errors import MissingSchema, PIDAlreadyExists
+from inspirehep.pidstore.minters.bai import BAIMinter
+
+
+def test_minter_bai(base_app, db, es, create_record):
+    data = {"ids": [{"schema": "JACOW", "value": "JACoW-12345678"}]}
+    record = create_record("aut", data=data, other_pids=["bai"])
+
+    expected_pids_len = 1
+    epxected_pids_values = [record["ids"][1]["value"]]
+    expected_pids_provider = "external"
+    expected_pids_status = PIDStatus.REGISTERED
+    result_pids = (
+        PersistentIdentifier.query.filter_by(object_uuid=record.id)
+        .filter_by(pid_type="bai")
+        .all()
+    )
+    result_pids_len = len(result_pids)
+
+    assert expected_pids_len == result_pids_len
+
+    pid = result_pids[0]
+    assert expected_pids_provider == pid.pid_provider
+    assert expected_pids_status == pid.status
+    assert pid.pid_value in epxected_pids_values
+
+
+def test_minter_orcid_empty(base_app, db, es, create_record):
+    record = create_record("aut")
+
+    expected_pids_len = 0
+
+    result_pids_len = PersistentIdentifier.query.filter_by(
+        object_uuid=record.id, pid_type="bai"
+    ).count()
+
+    assert expected_pids_len == result_pids_len
+
+
+def test_minter_orcid_already_existing(base_app, db, es, create_record):
+    record_with_orcid = create_record("aut", other_pids=["bai"])
+    data = {"ids": record_with_orcid["ids"]}
+    with pytest.raises(PIDAlreadyExists):
+        create_record("aut", data)
```

Comment:
IMO this is not testing the actual API it's a fixture, we should use Author record class for that

Reply:
We don't need to test Author record class, as we are interested only in Minter itself which is called correctly there and there are proper tests for this.

Reply:
but we might change the fixture, in that case we should either call minter create or the author class